instance/migrations/0042_add_instance_status.py

```diff
@@ -6,40 +6,37 @@
 from instance.models.instance import Status as InstanceStatus
 
 
+def get_current_server(instance):
+    return instance.server_set.order_by("id").last()
+
+def get_instance_state(server_status, server_progress):
+    if server_status in ("new", "active"):
+        return InstanceStatus.WaitingForServer.state_id
+    if server_status == "started":
+        if server_progress in ("running", "success"):
+            return InstanceStatus.WaitingForServer.state_id
+        elif server_progress == "failed":
+            return InstanceStatus.Error.state_id
+    if server_status in ("booted", "ready"):
+        return InstanceStatus.Running.state_id
+    if server_status == "provisioning":
+        if server_progress in ("running", "success"):
+            return InstanceStatus.ConfiguringServer.state_id
+        elif server_progress == "failed":
+            return InstanceStatus.ConfigurationFailed.state_id
+    if server_status == "rebooting":
+        return InstanceStatus.ConfiguringServer.state_id
+    if server_status == "terminated":
+        return InstanceStatus.Terminated.state_id
```

Comment:
Just a note that didn't occur to me before: It's better for migrations to not have any reference to the current state of the app, but we have some coupling here with `from instance.models.instance import Status as InstanceStatus`.

If we ever rename or remove one of these status types in the future, we'd need to modify this migration. So even though it feels wrong, it's usually better to just put string constants (`"waiting"` vs. `InstanceStatus.WaitingForServer.state_id`) in migrations.

I don't think it's worth making that change now though; we can easily cross that bridge when we come to it, if it ever even happens which is unlikely.